presto-main/src/main/java/io/prestosql/connector/informationschema/InformationSchemaMetadata.java

```diff
@@ -346,23 +353,23 @@ private boolean isTablesEnumeratingTable(SchemaTableName schemaTableName)
                             .filter(this::isLowerCase)
                             .map(table -> table.toLowerCase(ENGLISH))
                             .map(table -> new QualifiedObjectName(catalogName, prefix.getSchemaName().get(), table)))
-                    .filter(objectName -> metadata.getTableHandle(session, objectName).isPresent() || metadata.getView(session, objectName).isPresent())
+                    .filter(objectName -> !isColumnsEnumeratingTable(schemaTableName) || metadata.getTableHandle(session, objectName).isPresent() || metadata.getView(session, objectName).isPresent())
                     .filter(objectName -> !predicate.isPresent() || predicate.get().test(asFixedValues(objectName)))
                     .map(QualifiedObjectName::asQualifiedTablePrefix)
                     .collect(toImmutableSet());
         }
 
-        if (predicate.isPresent()) {
-            return prefixes.stream()
-                    .flatMap(prefix -> Stream.concat(
-                            metadata.listTables(session, prefix).stream(),
-                            metadata.listViews(session, prefix).stream()))
-                    .filter(objectName -> predicate.get().test(asFixedValues(objectName)))
-                    .map(QualifiedObjectName::asQualifiedTablePrefix)
-                    .collect(toImmutableSet());
+        if (!predicate.isPresent() || !isColumnsEnumeratingTable(schemaTableName)) {
```

Comment:
invert condition

Reply:
You already asked me to invert it this way in `InformationSchemaMetadata#calculatePrefixesWithSchemaName`. Let's keep it consistent.

Reply:
Then please put it as separate commit.